### ISSUE TITLE
liveQuery callbacks not always called (trying to do a count + group by)

### ISSUE THREAD

**Issue by justgage** [dexie/Dexie.js#1618]
We're trying to make a query that can take a bunch of notebooks and count how many entries they have. We came up with the following code. For the most part, it works but sometimes misbehaves and doesn't update the UI.

The code basically looks like this:
```ts
  subscribeToCounts(callback: (counts: Record<string, number>) => void) {
    const stream = liveQuery(async () => {
      const result: Record<string, number> = {};
      await Dexie.waitFor(
        this.db.entries
          .where(["is_deleted", "date", "journal_id", "id"])
          .between(
            [0, Dexie.minKey, Dexie.minKey, Dexie.minKey],
            [0, Dexie.maxKey, Dexie.maxKey, Dexie.maxKey],
            true,
            true
          )
          .eachKey((key) => {
            const journalId = (key as unknown as string[])[2];
            result[journalId] = result[journalId] ? result[journalId] + 1 : 1;
          })
      );
      console.log("NEW COUNT", result);
      return result;
    }).subscribe(
      (result) => {
        console.log("CALLING CALLBACK", callback, result);
        callback(result);
      },
      (err) => {
        console.error("Error in subscribeToCount", err);
      },
      () => {
        console.log("COMPLETED");
      }
    );
    return () => {
      stream.unsubscribe();
    };
  }
```

And we have a weird issue where `"CALLING CALLBACK"` doesn't get called very often as compared to `NEW COUNT`. Often times it seems to lag behind or not ever execute. I suspect this is because of how we're returning the `result`

Example of the logs
<img width="799" alt="Screen Shot 2022-09-19 at 2 07 28 PM" src="https://user-images.githubusercontent.com/164033/191112589-40424cf0-6bc7-48d9-a195-bbba489b41fb.png">

One way to resolve this is to put the `callback(result)` where `console.log("NEW COUNT", result);` is but that seems like the wrong thing to do.

**Comment by mrmurphy** [comment]
Bummer. You said this is not reliably reproducible @justgage ?

**Comment by dfahlander** [comment]
This is expected result if a mutation happens while the query is being executed. Since new mutations that can affect the query has occurred, the observable will not emit the obsolete result but instead reexecute the query.

**Comment by dfahlander** [comment]
Just got curiosity - why are you using Dexie.waitFor() in this case? That function is only useful when awaiting native promises (from things like webCrypto or fetch). In this case it probably won't hurt performance though as there's no transaction to keep alive but it's not needed.

**Comment by mrmurphy** [comment]
@dfahlander ah that makes sense. And this query is one of the most expensive we have, since it's doing a count/"group by" on the entire table.

**Comment by justgage** [comment]
@dfahlander yeah the `Dexie.waitFor()` was kind of a desperate thing I threw in at the last moment.

I think the problem is that count takes a decent amount of time and we're bulk inserting entries from the server (potentially thousands) along with decrypting them which can take a while. If I had to guess that it's basically so hammered it's not letting the count query finish.

I think what we need is more of a `throttle` than a `debounce` (emit values every so often even if they're technically stale). Also, we could put some artificial waits in there but I hate to slow down the initial sync since it can take a while as it is.
